src/main/java/net/sourceforge/jwbf/mediawiki/actions/queries/WatchList.java

```diff
@@ -0,0 +1,238 @@
+import com.google.common.primitives.Ints;
+
+public class WatchList extends BaseQuery<WatchResponse> {
+
+	public enum WatchListProperties {
+		//TODO: add all the properties
+		USER("user"), TITLE("title"), COMMENT("comment"), TIMESTAMP("timestamp"), PARSED_COMMENT(
+				"parsedcomment");
+
+		private String name;
+
+		private WatchListProperties(String name) {
+			this.name = name;
+		}
+
+		public String toString() {
+			return name;
+		}
+	}
+
+	public static final int DEFAULT_NS = 0;
+	private static final Joiner PARAM_JOINER = Joiner.on('|');
+
+	private static final Logger log = LoggerFactory.getLogger(WatchList.class);
+
+	private final ObjectMapper objectMapper = new ObjectMapper();
+
+	private final int limit;
+	private final ImmutableList<Integer> namespaces;
+	private WatchListProperties[] properties;
+	private boolean showBots;
+	private boolean showAnonymous;
+	private boolean showMinor;
+
+	private WatchListResults responseList;
+
+	private WatchList(Builder builder) {
+		super(builder.getBot());
+		if (!bot().isLoggedIn())
+			throw new ActionException("Please login first");
+		this.limit = builder.getLimit();
+		this.namespaces = ImmutableList.copyOf(Ints.asList(builder
+				.getNamespaces()));
+		this.properties = builder.getProperties();
+		this.showBots = builder.showBots;
+		this.showAnonymous = builder.showAnonymous;
+		this.showMinor = builder.showMinor;
+
+		initMapper();
+	}
+
+	private void initMapper() {
+		objectMapper.configure(
+				DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+
+	}
+
+	private String joinParam(Set<?> params) {
+		return MediaWiki.urlEncode(PARAM_JOINER.join(params));
+	}
+
+	@Override
+	protected WatchList copy() {
+		return new WatchList(bot());
+	}
+
+	@Override
+	protected HttpAction prepareNextRequest() {
+		RequestBuilder requestBuilder = new ApiRequestBuilder().action("query") //
+				.formatJson() //
+				.param("continue", MediaWiki.urlEncode("-||")) //
+				.param("list", "watchlist");
+		requestBuilder.param("wlnamespace",
+				MediaWiki.urlEncodedNamespace(namespaces)).param(
+				"wlprop",
+				joinParam(new HashSet<WatchListProperties>(Arrays
+						.asList(properties))));
+		if (showBots)
+			requestBuilder.param("wlshow", "bot");
+		else
+			requestBuilder.param("wlshow", "!bot");
+		if (showAnonymous)
+			requestBuilder.param("wlshow", "anon");
+		else
+			requestBuilder.param("wlshow", "!anon");
+		if (showMinor)
+			requestBuilder.param("wlshow", "minor");
+		else
+			requestBuilder.param("wlshow", "!minor");
+		if (limit < 1)
+			requestBuilder.param("wllimit", "max");
+		else
+			requestBuilder.param("wllimit", limit);
+		if (hasNextPageInfo()) {
+			requestBuilder.param("wlcontinue", getNextPageInfo());
+		}
+		log.debug("using query {}", requestBuilder.build());
+		return requestBuilder.buildGet();
+
+	}
+
+	@Override
+	protected ImmutableList<WatchResponse> parseElements(String json) {
+		try {
+			this.responseList = objectMapper.readValue(json,
```

Comment:
try to use a ```JsonMapper```; for examples see ```JsonMapperTest```. It is a wrapper for which converts checked to unchecked exceptions (see comment above)